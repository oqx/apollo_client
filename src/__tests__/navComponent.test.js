import React from "react";
import ConnectedNav from "../components/navComponent";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";

const mockStore = configureStore();
let store, container, mountedContainer;

const initialState = {
  ui_reducer: {
    sidebar_state: true
  }
};

let state = {
  btnClick: false
};

const spy = jest.fn();

const handleClickStub = () => {
  state.btnClick = true;
  spy();
};

describe("<Nav />", () => {
  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedNav store={store} />).dive();
  });

  it("should render successfully", () => {
    expect(container.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(toJson(container)).toMatchSnapshot();
  });
});
