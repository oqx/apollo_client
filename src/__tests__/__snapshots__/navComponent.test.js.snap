// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match snapshot 1`] = `
<nav
  className="nav"
>
  <div
    className="nav__logo-col"
  >
    <Component
      marginRight="12px"
    />
    <Component />
  </div>
  <div
    className="nav__links-col"
  >
    <ul
      className="ul ul--horizontal"
    >
      <li
        className="ul__li"
      >
        <Component
          controlFunc={[Function]}
          iconClass="ion-ios-search"
        />
      </li>
    </ul>
  </div>
</nav>
`;
